File: 프로그래머스/lv2/17683. ［3차］ 방금그곡/［3차］ 방금그곡.py
def diffMinute(sTime, eTime):
    sh, sm = map(int, sTime.split(":"))
    eh, em = map(int, eTime.split(":"))
    return (eh*60 + em) - (sh*60 + sm)

# ABC와 ABC# 조심
def solution(m, musicinfos):
    answer = '(None)'
    maxMinute = 0
    m = m.replace('C#', 'c').replace('D#', 'd').replace('F#', 'f').replace('G#', 'g').replace('A#', 'a')
    
    for music in musicinfos:
        sTime, eTime, title, code = music.split(",")
        code = code.replace('C#', 'c').replace('D#', 'd').replace('F#', 'f').replace('G#', 'g').replace('A#', 'a')
        diff = diffMinute(sTime, eTime)
        playCode = ''
        for i in range(diff):
            playCode += code[i%len(code)]
        if m in playCode:
            if diff > maxMinute:
                maxMinute = diff
                answer = title
        
    return answer

File: 프로그래머스/lv2/17683. ［3차］ 방금그곡/［3차］ 방금그곡.java
class Solution {
    public String solution(String m, String[] musicinfos) {
        String answer = "(None)";
        int maxMinute = 0;
        m = m.replace("C#", "c").replace("D#", "d").replace("F#", "f").replace("G#", "g").replace("A#", "a");
        
        for(String muisc : musicinfos) {
            String[] info = muisc.split(",");
            String title = info[2];
            String code = info[3].replace("C#", "c").replace("D#", "d").replace("F#", "f").replace("G#", "g").replace("A#", "a");
            int diff = diffTime(info[0], info[1]);
            StringBuilder playCode = new StringBuilder();
            for(int i = 0; i<diff; i++){
                playCode.append(code.charAt(i%code.length()));
            }
            
            if(playCode.toString().contains(m) && diff > maxMinute){
                    maxMinute = diff;
                    answer = title;
            }
        }
        
        return answer;
    }
    
    public int diffTime(String sTime, String eTime){
        String[] st = sTime.split(":");
        String[] et = eTime.split(":");
        int sh = Integer.parseInt(st[0]);
        int sm = Integer.parseInt(st[1]);
        int eh = Integer.parseInt(et[0]);
        int em = Integer.parseInt(et[1]);
        return (eh*60 + em) - (sh*60 + sm);
    }
}